### PR TITLE
Autocompletion suggestions in global search

### LINKED ISSUE #51: Improve geocoder
- [ ] Add autocomplete
- [ ] Support Long-Lat coordinates

### PR DESCRIPTION
**Pull request by tombh**
This is a working version. But what do other people think about the UX for going to a coordinate? I've removed the search button, so you can only go to suggestions in the dropdown (if there's nothing in the dropdown then there's nowhere to go in the world anyway). So it can feel a bit weird sometimes when you're typing a coord and suggestions for real places (that are near that coord) start appearing.

Includes the addition of `react-autocomplete` package.

Also I refactored search into its own component and started using
ES7 classes, which meant adding `babel-eslint` and
`babel-plugin-transform-class-properties`.

Fixes #51

### CODE REVIEW

**Comment by smit1678** [comment]
@tombh Really nice work. Just testing it out this morning. Couple comments on the UX side of things: 

> But what do other people think about the UX for going to a coordinate?

Agreed, it is a bit weird. Looking at some other implementations, perhaps geojson.io has some good insight? Searches are only fired after you press `enter` or click the search icon: https://monosnap.com/file/F5Ip3PU0BJjutp1cYb0PpadXha4HOX. 

If we wait to search until some presses enter, perhaps that's better UX? Then someone can complete adding coordinates and then search. 

I'm also thinking that we need to keep the search icon for usability. That's a helpful marker to know that pressing enter or clicking the button is the method for interacting with the search.

**Comment by tombh** [comment]
Thanks Nate. Oh yeah I forgot that the search icon is useful for that! I'll definitely put that back and the geojson.io UX really makes sense as well. So let me do that ...

**Comment by tombh** [comment]
I pushed an update with the previously mentioned fixes.

File: app/assets/styles/_structure.scss [hunk 1]
```diff
@@ -96,11 +96,26 @@
...
+    }
+  }
 }
 
 .global-search__button-go {
   @extend .button, .button--achromic, .button--text-hidden, .button--medium;
 
+  // TODO: Absolute positioning is not good
+  position: relative;
+  top: -3px;
```
Comment (tombh) [inline app/assets/styles/_structure.scss]: I don't like this, but I was taking so long figuring out a better way :/
Any ideas @danielfdsilva?

Comment (danielfdsilva) [inline app/assets/styles/_structure.scss]: @tombh Maybe it has to do with the vertical alignment. However @ricardomestre would be better suited to help here

Comment (tombh) [inline app/assets/styles/_structure.scss]: Thanks Daniel, I'll see if Ricardo has anything to say as well then.

**Comment by tombh** [comment]
Cooool, thank you @ricardomestre :)

**Pull request merged**